file-processing/README.md

```diff
@@ -0,0 +1,304 @@
+# File processing
+
+## Overview
+
+In this tutorial, you will develop an application that uses a directory listener to pick from a directory new files added with a specific extension (e.g., `.csv`), process the file content, insert records to a database based on the file content, and then move the file to another directory. 
+
+To implement this use case, you will develop a file service with a single remote method using Visual Studio Code with the Ballerina Swan Lake extension. The remote method will be called whenever a new file is added to the particular directory.
+
+The flow is as follows.
+
+1. When a new file is added to the directory, pick it for further processing if it is a CSV file (if the extension is `.csv`).
+
+2. If it is a CSV file, read the file content which is expected to be in the following form.
+
+    ```csv
+    First Name, Last Name, Phone
+    Amy, Roy, 0112222222
+    Joy, Williams, 0111111111
+    ```
+
+3. Insert the read records to a database. Insert them to a table named `Persons` of which the column names are expected to be `firstName`, `lastName`, and `phone`.
+
+4. If successful, move the file to a directory containing successfully-processed files. If processing failed in step 2 or 3, move the file to a directory containing files for which processing failed.
+
+### Concepts covered
+
+- File listener
+- File operations
+- SQL/MySQL clients
+- CSV files
+
+## Develop the application
+
+### Step 1: Set up the workspace
+
+Install [Ballerina Swan Lake](https://ballerina.io/downloads/) and the [Ballerina Swan Lake VS Code extension](https://marketplace.visualstudio.com/items?itemName=wso2.ballerina) on VS Code.
+
+### Step 2: Develop the service
+
+Follow the instructions given in this section to develop the service.
+
+1. Create a new Ballerina project using the `bal` command and open it in VS Code.
+
+    ```bash
+    $ bal new file-processing
+    ```
+
+2. Introduce the source code in files with the `.bal` extension (e.g., the `main.bal` file). Introduce a constant to represent the `.csv` extension and [configurable variables](https://ballerina.io/learn/by-example/#configurability) to denote the paths to the three directories: directory to listen on and directories to move files to once processing is completed. Also introduce configurable variables for database configuration.
+
+    ```ballerina
+    const CSV_EXT = ".csv";
+
+    configurable string inPath = "./in";
+    configurable string mvOnSuccessPath = "./out";
+    configurable string mvOnFailurePath = "./failed";
+    configurable string host = ?;
+    configurable string user = ?;
+    configurable string password = ?;
+    configurable string database = ?;
+    configurable int port = ?;
```

Comment:
Why do we need this change? The current order is the same as the parameter order.

Reply:
Ah, I see, Since we are using this pattern that's fine. We don't need this suggestion. I thought it would be better to have an order like: host, port, db and then dbuser credentials